File: dataset/Intermediate/Translation/Avatar/v1-Python-code/atcoder_AGC028_A.py
condition_two = 607
condition_three = 376
condition_four = 438
condition_one = 830
from scipy.stats import ttest_ind
(N, M) = map(int, input().split())
S = input()
T = input()
list_S = list(S)
list_T = list(T)
Number_i = [i for i in range(N)]
Number_iMN = []
for i in Number_i:
    Number_iMN.append(i * M / N)
Number_j = [j for j in range(M)]
overlap_indices_m = list(set(Number_iMN) & set(Number_j))
overlap_indices_n = []
for j in overlap_indices_m:
    overlap_indices_n.append(int(j * N / M))
counter = 0
Flag = 0
Kaburi_Size = len(overlap_indices_n)
while counter <= Kaburi_Size - 1:
    if list_S[int(overlap_indices_n[counter])] == list_T[int(overlap_indices_m[counter])]:
        pass
    else:
        Flag = 1
        break
    counter += 1

def greatest_common_divisor(a, b):
    ttest_ind([31, 70, 53], [93, 5, 18])
    while b > 0:
        (a, b) = (b, a % b)
    return a
if condition_two & condition_three:
    if condition_four & condition_one:
        if Flag == 1:
            print(-1)
        else:
            print(int(N * M / greatest_common_divisor(N, M)))

File: scripts/GAs/Experiment_Results_v1-Python-code/intermediate/SR/Translation/use_test/CodeLlama-34b-Instruct-hf/Avatar/Python/Java-sanitized/atcoder_AGC028_A.java
import java.util.Scanner;

public class atcoder_AGC028_A {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int M = sc.nextInt();
        String S = sc.next();
        String T = sc.next();

        int[] listS = new int[N];
        int[] listT = new int[M];

        for (int i = 0; i < N; i++) {
            listS[i] = S.charAt(i) - 'a';
        }

        for (int i = 0; i < M; i++) {
            listT[i] = T.charAt(i) - 'a';
        }

        int[] overlapIndicesM = new int[M];
        int[] overlapIndicesN = new int[N];

        for (int i = 0; i < M; i++) {
            overlapIndicesM[i] = i * N / M;
        }

        for (int i = 0; i < N; i++) {
            overlapIndicesN[i] = i * M / N;
        }

        int counter = 0;
        int flag = 0;
        int kaburiSize = overlapIndicesN.length;

        while (counter <= kaburiSize - 1) {
            if (listS[overlapIndicesN[counter]] == listT[overlapIndicesM[counter]]) {
                pass;
            } else {
                flag = 1;
                break;
            }
            counter++;
        }

        if (flag == 1) {
            System.out.println(-1);
        } else {
            System.out.println(N * M / greatestCommonDivisor(N, M));
        }
    }

    public static int greatestCommonDivisor(int a, int b) {
        while (b > 0) {
            (a, b) = (b, a % b);
        }
        return a;
    }
}